lib/galaxy/webapps/galaxy/services/datasets.py

```diff
@@ -929,7 +929,7 @@ def _data(
 
             # Get mean depth.
             if not indexer:
-                indexer = self.data_provider_registry.get_data_provider(trans, original_dataset=dataset, source="index")
+                indexer = self._get_indexer(trans, dataset)
```

Comment:
Minor thing for clarity here: Can you drop `indexer = None` from line 866 and drop the `if not indexer:` guard here ?

Reply:
Done!